shaka/src/js/xml_http_request.cc

```diff
@@ -301,6 +302,21 @@ ExceptionOr<void> XMLHttpRequest::SetRequestHeader(const std::string& key,
   return {};
 }
 
+bool XMLHttpRequest::WithCredentials() const {
+  return with_credentials_;
+}
+ 
+ExceptionOr<void> XMLHttpRequest::SetWithCredentials(bool with_credentials) {
+  if (ready_state != XMLHttpRequest::ReadyState::Unsent &&
+      ready_state != XMLHttpRequest::ReadyState::Opened) {
+    return JsError::DOMException(InvalidStateError, "Failed to set the 'withCredentials' property on 'XMLHttpRequest': The value may only be set if the object's state is UNSENT or OPENED.");
```

Comment:
nit: Line too long.  Reduce to 80 characters.  Also, I think you should remove the prefix here.